server/auth/middleware.go

```diff
@@ -0,0 +1,252 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/bitcoin-sv/spv-wallet/config"
+	"github.com/bitcoin-sv/spv-wallet/engine"
+	"github.com/bitcoin-sv/spv-wallet/engine/utils"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/gin-gonic/gin"
+	"github.com/libsv/go-bt/v2/bscript"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// ParamXPubKey the request parameter for the xpub string
+	ParamXPubKey = "xpub"
+
+	// ParamXPubHashKey the request parameter for the xpub ID
+	ParamXPubHashKey = "xpub_hash"
+
+	// ParamAccessKey the request parameter for the xpub ID
+	ParamAccessKey = "access_key"
+
+	// ParamAdminRequest the request parameter whether this is an admin request
+	ParamAdminRequest = "auth_admin"
+
+	// ParamAuthSigned the request parameter that says whether the request was signed
+	ParamAuthSigned = "auth_signed"
+)
+
+// Payload is the authentication payload for checking or creating a signature
+type Payload struct {
+	AuthHash     string `json:"auth_hash"`
+	AuthNonce    string `json:"auth_nonce"`
+	AuthTime     int64  `json:"auth_time"`
+	BodyContents string `json:"body_contents"`
+	Signature    string `json:"signature"`
+	xPub         string
+	accessKey    string
+}
+
+// CorsMiddleware is a middleware that handles CORS.
+func CorsMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+		c.Writer.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS")
+		c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type, Cache-Control"+", "+models.AuthHeader+", "+models.AuthAccessKey+", "+models.AuthSignature+", "+models.AuthHeaderHash+", "+models.AuthHeaderNonce+", "+models.AuthHeaderTime)
```

Comment:
It would be more readable and easier to maintain if you would do something like:
```
	corsAllowedHeaders := []string{
		"Content-Type",
		"Cache-Control",
		models.AuthHeader,
                 ....
	}
	c.Writer.Header().Set("Access-Control-Allow-Headers", strings.Join(corsAllowedHeaders, ","))
```

See here example:
https://go.dev/play/p/Av7tCM2aOya

Reply:
Done